astroid/__init__.py

```diff
@@ -107,8 +107,22 @@ def inference_tip(infer_function):
 
        MANAGER.register_transform(Call, inference_tip(infer_named_tuple),
                                   predicate)
+
+    .. Note::
+
+        ALWAYS use a predicate with an inference_tip
+        since registering a transform will
+        influence inference: an interance tip
+        overwrites the _explicit_inference
+        attribute
     """
     def transform(node, infer_function=infer_function):
+        if (raise_on_overwrite
+                and node._explicit_inference is not None
+                and node._explicit_inference is not infer_function):
+            raise Exception(
```

Comment:
Let's raise a custom astroid exception, by adding one in `astroid/exceptions.py`